pkg/beholder/auth.go

```diff
@@ -11,17 +13,81 @@ var authHeaderKey = "X-Beholder-Node-Auth-Token"
 // authHeaderVersion is the version of the auth header format
 var authHeaderVersion = "1"
 
+// BuildAuthHeadersOpt are used to modify the behavior of the BuildAuthHeaders function
+type BuildAuthHeadersOpt func(*authHeaderConfig)
+
+type authHeaderConfig struct {
+	privKey   ed25519.PrivateKey
+	timestamp int64
+	version   string
+}
+
 // BuildAuthHeaders creates the auth header value to be included on requests.
-// The current format for the header is:
+// There are two formats for the header. Version `1` is:
 //
 // <version>:<public_key_hex>:<signature_hex>
 //
 // where the byte value of <public_key_hex> is what's being signed
-func BuildAuthHeaders(privKey ed25519.PrivateKey) map[string]string {
+// and <signature_hex> is the signature of the public key.
+// The version `2` is:
+//
+// <version>:<public_key_hex>:<timestamp>:<signature_hex>
+//
+// where the byte value of <public_key_hex> and <timestamp> are what's being signed
+func BuildAuthHeaders(privKey ed25519.PrivateKey, opts ...BuildAuthHeadersOpt) map[string]string {
```

Comment:
Is there a compelling reason to use functional options? I am worried about how they pollute the package API namespace with verbose functions. Would something simple and direct suffice?
```suggestion
func (AuthHeadersConfig) Build(privKey ed25519.PrivateKey) map[string]string {
```

Reply:
Allows existing backwards compatibility of `BuildAuthHeaders`. But I agree,  I think the most direct approach here would be to expose `BuildAuthHeadersV2`

Reply:
`BuildAuthHeadersV2` sounds good, but FWIW adding variadic args may not break callers, but it does change the function signature and is not compatible by definition. This article is a good read: https://go.dev/blog/module-compatibility#adding-to-a-function
> Sometimes the techniques of adding a new function and adding options can be combined by making the options struct a method receiver. Consider the evolution of the net package’s ability to listen at a network address.

https://pkg.go.dev/net@go1.11#ListenConfig

Reply:
Thanks ^ this was helpful, I took a look at the `crypto/tls` `Dial` example